src/modules/lua_mtev.c

```diff
@@ -1155,18 +1204,40 @@ mtev_lua_socket_read(lua_State *L) {
   if(args == 1) return 1; /* completed read, return result */
   if(len == -1 && errno == EAGAIN) {
     /* we need to drop into eventer */
-  }
+    eventer_remove_fd(e->fd);
+    e->callback = mtev_lua_socket_read_complete;
+    e->mask = mask | EVENTER_EXCEPTION;
+    eventer_add(e);
+
+    if (lua_gettop(L) == 5 && lua_isfunction(L, 5)) {
+      int timeout;
+      timeout = 10;
+      if(lua_isnumber(L, 4)) {
+        timeout = lua_tointeger(L, 4);
```

Comment:
We should allow a double here representing sub-second values.

There are a lot of times when a read might want a 100ms timeout. "0.1"